Wildcard subscription doc (#1971)
emqtt/emq-docs-cn: en_US/messaging/mqtt-wildcard-subscription.md

```
"+" is valid
"sensor/+" is valid
"sensor/+/temperature" is valid
"sensor+" is invalid (does not occupy an entire level)
```

If the client subscribes to the topic `sensor/+/temperature`, it will receive messages from the following topics:

```awk
sensor/1/temperature
sensor/2/temperature
...
sensor/n/temperature
```

But it will not match the following topics:

```bash
sensor/temperature
sensor/bedroom/1/temperature
```

## Multi-Level Wildcard

`#` (U+0023) is a wildcard character that matches any number of levels within a topic. When using a multi-level wildcard, it must occupy an entire level and must be the last character of the topic, for example:

```pgsql
"#" is valid, matches all topics
"sensor/#" is valid
"sensor/bedroom#" is invalid (+ or # are only used as a wildcard level)
"sensor/#/temperature" is invalid (# must be the last level)
```

> ## Try Wildcard  Subscription with MQTTX Client

Reviewer: ```suggestion
## Try Wildcard Subscription with MQTTX Client
```

> This section demonstrates how to use the MQTTX client to simulate subscriptions to wildcard topics. 

Reviewer: ```suggestion
This section demonstrates how to use the MQTTX client to create subscriptions to wildcard topics. 
```

> In this demonstration, you can create one client connection `demo` as a publisher to publish messages. Then, you can create another client connection as a subscriber. The subscriber subscribes to the following wildcard topics:

Reviewer: The two client IDs in the screenshot, one with the first letter capitalized and one without.

Author: Changed to "Demo" throughout.

- `testtopic/+/temperature`
- `testtopic/#`

1. Start the MQTTX Client. Click the **New Connection** to create an MQTT connection named `Demo`.

   - The localhost `127.0.0.1` is used as an example in this demonstration.

   ::: tip

   For detailed instructions on creating an MQTT connection, see [MQTTX Client](./publish-and-subscribe.md).

   :::

   <img src="/Users/emqx/Documents/GitHub/emqx-docs/en_US/messaging/assets/Configure-new-connection-general.png" alt="Configure-new-connection-general" style="zoom:35%;" />

2. Click **+** in the **Connections** pane to create another connection as a subscriber. Set **Name** to `Subscriber`.

3. Select `Subscriber` in **Connections** and click **+ New Subscription**. In the pop-up dialogue, input `testtopic/+/temperature` in the **Topic** field. Leave other options as default.

   <img src="./assets/wildcard-sub-1.png" alt="wildcard-sub-1" style="zoom: 38%;" />

> 4. Select `Demo` in **Connections**. Type `testtopic/room1/temperature` in the topic field and type the message payload `28 degree` in the message field. Click the send button. Send another message with the topic `testtopic/room2/temperature`.

Reviewer: ```suggestion
4. Select `Demo` in **Connections**. Type `testtopic/room1/temperature` in the topic field and type the message payload `28 degree` in the message field. Click the send button. Send another message with the same payload to the topic `testtopic/room2/temperature`.
```

      <img src="./assets/wildcard-sub-2.png" alt="wildcard-sub-2" style="zoom:40%;" />

> 5. Select `Subscriber` in **Connections**. You can see the subscriber receives the two messages the publisher sends.

Reviewer: ```suggestion
5. Select `Subscriber` in **Connections**. You can see the subscriber receives the two messages sent by the publisher.
```
